src/coord/axisCommonTypes.ts

```diff
@@ -179,6 +179,8 @@ interface AxisLineOption {
 
 interface AxisTickOption {
     show?: boolean | 'auto',
+    alignWithLabel?: boolean,
+    interval?: 'auto' | number | ((index:number, value: string) => boolean),
```

Comment:
Hi, thanks for your contribution! Since `alignWithLabel` and `interval` only work for the category axis, we put these two options into [`CategoryAxisBaseOption`](https://github.com/apache/echarts/blob/master/src/coord/axisCommonTypes.ts#L139-L143) rather than the common `AxisTickOption`.

/cc @pissang

Reply:
Ah, okay. We had been using `XAxisOption` and I had considered `SingleAxisComponentOption` as we had logic to toggle between `value` and `category` depending on incoming data and that had felt appropriate. I suppose we could get by by doing `CategoryAxisBaseOptions & ValueAxisBaseOption` (as opposed to doing the intersection that `XAxisOption` is doing.